frontend/src/features/admin/agencies/ManageAgencies.tsx

```diff
@@ -0,0 +1,112 @@
+import './ManageAgencies.scss';
+import React, { useMemo, useCallback, useState } from 'react';
+import { Container } from 'react-bootstrap';
+import { Table } from 'components/Table';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from 'reducers/rootReducer';
+import { columnDefinitions } from '../constants';
+import { IAgenciesState } from 'reducers/agencyReducer';
+import { IAgency, IAgencyDetail, IAgencyFilter, IAgencyRecord, IPagedItems } from 'interfaces';
+import { getAgenciesAction } from 'actionCreators/agencyActionCreator';
+import { toFilteredApiPaginateParams } from 'utils/CommonFunctions';
+import { TableSort } from 'components/Table/TableSort';
+import { IGenericNetworkAction } from 'actions/genericActions';
+import * as actionTypes from 'constants/actionTypes';
+import { generateSortCriteria } from 'utils';
+import { AgencyFilterBar } from './AgencyFilterBar';
+import useCodeLookups from 'hooks/useLookupCodes';
+import { useHistory } from 'react-router-dom';
+
+const ManageAgencies: React.FC<any> = ({ filterable, title, mode }) => {
+  const columns = useMemo(() => columnDefinitions, []);
+  const dispatch = useDispatch();
+  const [filter, setFilter] = useState<IAgencyFilter>({});
+  const lookupCodes = useCodeLookups();
```

Comment:
This kind of defeats the purpose of the server side paging... but I was not sure how to get the string `label `for the parent without it as I can't match up the `parentId `with the `id `without it. I cannot compare it against the paged agencies as they do not contain all the potential parents, should I work on implementing something on the backend? Or is this ok.

Reply:
You could do this a few different ways.  You could update the models when fetching agencies to include some minor details about their parent, or vise-versa.  The way you are doing will only work until they add or rename an agency, as the lookup will become invalid.

Reply:
Which highlights an issue that will appear to be a bug for some once someone does add/edit/delete an agency.

Reply:
Gotchya, thank you!

Reply:
Not to throw a wrench into this, but do we need server side paging at all here? this seems like a relatively small (kb) dataset that shouldn't grow much over time.

Reply:
Yeah I talked about this with Jeremy briefly after I had realized it may be unnecessary.. we decided on it won't hurt anything but in the future for small data sets always do client side